### PR TITLE
Feature/dsci 50 fix backend search tests

### PR DESCRIPTION
**Pull request by kdutia**
Fix some mypi errors (there are more to go...), and add some quality of life fixes to the indexing process: 

* `--limit` flag used for debug now limits the number of descriptions to just those that are in the documents the limited text is from
* increased timeout for Opensearch indexing

### CODE REVIEW

File: app/index.py [hunk 1]
```diff
@@ -204,7 +204,7 @@ def bulk_index(self, actions: Iterable[dict]):
         successes = 0
 
         for ok, _ in helpers.streaming_bulk(
-            client=self.opns, index=self.index_name, actions=actions
+            client=self.opns, index=self.index_name, actions=actions, timeout="30s"
```
Comment (diversemix) [inline app/index.py:207]: "magic" number here - is this worth pulling out into config at some stage?

File: text2embeddings.py [hunk 2]
```diff
@@ -56,8 +56,8 @@ def get_text_from_list(text_block: dict, prev_processed_text_block: dict) -> str
     #  when, for example, there are random elements between pages that the postprocessors haven't been able to catch.
     #  But I've ignored this for now because it seems that the postprocessing in postprocessor.py has dealt with the
     #  vast majority of edge cases (as indicated by the rarity of certain debugging metadata).
-    if prev_processed_text_block:
-        text = prev_processed_text_block + "\n" + text
+    if prev_processed_text:
```
Comment (diversemix) [inline text2embeddings.py:59]: Consider:
```
return text if not prev_processed_text else prev_processed_text + "\n" + text
```

**Review by diversemix — APPROVED** [review]
Good first step toward mypi heaven :1st_place_medal:

File: app/index.py (hunk 1 above)
Comment (kdutia) [reply app/index.py]: done!

**Pull request merged**